lib/core/src/persist/mod.rs

```diff
@@ -194,18 +214,32 @@ impl Persister {
             .optional()?)
     }
 
-    pub fn get_payments(&self) -> Result<HashMap<String, Payment>> {
+    pub fn get_payments(&self) -> Result<Vec<Payment>> {
         let con = self.get_connection()?;
 
         // Assumes there is no swap chaining (send swap lockup tx = receive swap claim tx)
         let mut stmt = con.prepare(&self.select_payment_query(None))?;
-
-        let data = stmt
-            .query_map(params![], |row| {
-                self.sql_row_to_payment(row).map(|p| (p.tx_id.clone(), p))
-            })?
+        let payments: Vec<Payment> = stmt
+            .query_map(params![], |row| self.sql_row_to_payment(row))?
             .map(|i| i.unwrap())
             .collect();
-        Ok(data)
+        let unclaimed_payments = self
+            .list_unclaimed_pending_receive_swaps()?
+            .into_iter()
+            .map(|pending_receive_swap| Payment {
+                tx_id: None,
+                swap_id: Some(pending_receive_swap.id),
+                timestamp: pending_receive_swap.created_at,
+                amount_sat: pending_receive_swap.receiver_amount_sat,
+                fees_sat: pending_receive_swap.payer_amount_sat
+                    - pending_receive_swap.receiver_amount_sat,
+                preimage: None,
+                refund_tx_id: None,
+                refund_tx_amount_sat: None,
+                payment_type: PaymentType::Receive,
+                status: pending_receive_swap.state,
+            })
+            .collect();
+        Ok([payments, unclaimed_payments].concat())
```

Comment:
The fact we are concatenating here will prevent us from exposing paging/filtering interface for the list_payments API.
I think we should come up with a way to get the pending receives as part of the `select_payment_query`
Perhaps we can use full outer join instead?

Reply:
Since this flow is getting more complicated now I suggest to add some tests to the `get_payments` function with various payment types.